client_internals_it_test.go

```diff
@@ -383,6 +383,36 @@ func TestClientInternal_EncodeData(t *testing.T) {
 	})
 }
 
+func TestProxyManagerShutdown(t *testing.T) {
+	clientTester(t, func(t *testing.T, smart bool) {
+		ctx := context.Background()
+		tc := it.StartNewClusterWithOptions("proxy-manager-graceful-shutdown", 5701, 1)
+		defer tc.Shutdown()
+		config := tc.DefaultConfig()
+		config.Cluster.Unisocket = !smart
+		client := it.MustClient(hz.StartNewClientWithConfig(ctx, config))
```

Comment:
Could you defer shutdown the client?

Reply:
Resolved, [0042502](https://github.com/hazelcast/hazelcast-go-client/pull/772/commits/0042502d165af4bbfd57b99474f816dd147c4599).